# CAR-SALES dataset

Dữ liệu `car-sales.csv`: bộ dữ liệu chứa thông tin các ôtô cũ ở một cửa hàng bán ô tô cũ. 

Bạn hãy thực hiện các yêu cầu sau:


### 1. Đọc dữ liệu và cho biết dữ liệu có bao nhiêu dòng, bao nhiêu cột

Đáp án: (10 dòng, 5 cột)

In [1]:
import pandas as pd
import numpy as np

In [2]:
car = pd.read_csv('.\car-sales.csv')


In [3]:
car.shape
# 10 dong, 5 cot

(10, 5)

In [4]:
car

Manufacturer Colour  Odometer  Doors    Price
0       Toyota  White    150043      4   4000.0
1        Honda    Red     87899      4   5000.0
2       Toyota   Blue     32549      3   7000.0
3          BMW  Black     11179      5  22000.0
4       Nissan  White    213095      4   3500.0
5       Toyota  Green     99213      4   4500.0
6        Honda   Blue     45698      4   7500.0
7        Honda   Blue     54738      4   7000.0
8       Toyota  White     60000      4   6250.0
9       Nissan  White     31600      4   9700.0

### 2. Cho biết thuộc tính nào là thuộc tính phân loại, thuộc tính nào là thuộc tính định lượng

In [5]:
# Your code here
car.info()
# Manufacturer and Colour: dinh luong
# Còn lại định tính

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10 entries, 0 to 9
Data columns (total 5 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Manufacturer  10 non-null     object 
 1   Colour        10 non-null     object 
 2   Odometer      10 non-null     int64  
 3   Doors         10 non-null     int64  
 4   Price         10 non-null     float64
dtypes: float64(1), int64(2), object(2)
memory usage: 528.0+ bytes


### 3. Trong dữ liệu có thông tin của bao nhiêu xe?

Đáp án: 10 xe

In [6]:
car['Manufacturer'].count()

10

In [7]:
car.count()

Manufacturer    10
Colour          10
Odometer        10
Doors           10
Price           10
dtype: int64

### 4. Đếm số xe theo theo hãng

Đáp án: Toyota: 10 xe, Honda: 3 xe, Nissan: 2 xe, BMW: 1 xe

In [8]:
# Your code here
car['Manufacturer'].value_counts()

Manufacturer
Toyota    4
Honda     3
Nissan    2
BMW       1
Name: count, dtype: int64

### 5. Đếm số xe theo màu


In [9]:
# Your code here
car['Colour'].value_counts()

Colour
White    4
Blue     3
Red      1
Black    1
Green    1
Name: count, dtype: int64

### 6. Tổng giá trị các xe trong cửa hàng

Đáp án: 76450 USD


In [10]:
# Your code here
a = int(car['Price'].sum())
b = str(a)
print(b+ " USD")


76450 USD


### 7. Top 3 xe có giá cao nhất


In [19]:
# Your code here
a = car['Price'].sort_values(ascending=False).head(3)
print(a)
# CÁCH chuan
b = car.sort_values(by='Price',ascending=True).head(3)
print(b)

3    22000.0
9     9700.0
6     7500.0
Name: Price, dtype: float64


### 8. Top 3 xe đã chạy nhiều cây số (km)nhất (Odometer)

Đáp án:
- Xe Nissan, White, 213095 km
- Xe Toyota, White, 150043 km
- Xe Toyota, Green, 99213 km

In [53]:
# Your code here
car.sort_values(by='Odometer',ascending=False).head(3)

Manufacturer Colour  Odometer  Doors   Price
4       Nissan  White    213095      4  3500.0
0       Toyota  White    150043      4  4000.0
5       Toyota  Green     99213      4  4500.0

### 9. Khảo sát xe của hãng Honda

- Tính tỷ lệ xe Honda trong dữ liệu của cửa hàng (0.3)
- Cho biết giá trung bình của xe hãng Honda của cửa hàng (6500)
- Cho biết các xe của hãng Honda trong cửa hàng có những màu sắc nào (Red, Blue)

In [117]:
# Your code here
b=car[car.Manufacturer=='Honda']
a=len(car)
print('Tính tỷ lệ xe Honda trong dữ liệu của cửa hàng',len(b)/a)
print(b)
print("Cho biết giá trung bình của xe hãng Honda của cửa hàng", str(int(b['Price'].mean())))
c = b['Colour']
print("Cho biết các xe của hãng Honda trong cửa hàng có những màu sắc nào (Red, Blue)\n",c.unique())


Tính tỷ lệ xe Honda trong dữ liệu của cửa hàng 0.3
  Manufacturer Colour  Odometer  Doors   Price
1        Honda    Red     87899      4  5000.0
6        Honda   Blue     45698      4  7500.0
7        Honda   Blue     54738      4  7000.0
Cho biết giá trung bình của xe hãng Honda của cửa hàng 6500
Cho biết các xe của hãng Honda trong cửa hàng có những màu sắc nào (Red, Blue)
 ['Red' 'Blue']


# 10. Tính toán các giá trị sau của  thuộc tính giá ('Price')

- Giá trung bình các xe (7645)
- Giá trị trung vị các xe (6625)
- Miền giá trị của xe (18500)
- Miền phân vị của xe (2750)
- Điểm phân vị 0.4 và 0.6 của giá ([5750, 7000])

In [116]:
# Your code here
print("Giá trung bình các xe: ",car['Price'].mean())
print("Giá trị trung vị các xe:", car['Price'].median())
print("Miền giá trị của xe:", car['Price'].max()-car['Price'].min())

a=car['Price'].quantile(75/100) 
b=car['Price'].quantile(25/100) 
print("Miền phân vị của xe", a-b)

c=car['Price'].quantile(0.4) 
d=car['Price'].quantile(0.6) 
e = []
e.append(c)
e.append(d)
# print([c,d])
print("Điểm phân vị 0.4 và 0.6 của giá", e)


Giá trung bình các xe:  7645.0
Giá trị trung vị các xe: 6625.0
Miền giá trị của xe: 18500.0
2750.0
Điểm phân vị 0.4 và 0.6 của giá [5750.0, 7000.0]


---

# TIPS dataset

Dữ liệu chứa trong file `tips.csv` chứa thông tin tiền tip của khách hàng tại một nhà hàng. Dữ liệu có các thông tin sau:

- `total_bill`: tổng hoá đơn (USD)
- `tip`: tiền tip/ tiền boa (USD)
- `sex`: giới tính người thanh toán hóa đơn
- `smoker`: có hút thuốc hay không
- `day`: ngày trong tuần
- `time`: thời gian dùng bữa(buổi sáng, buổi trưa, buổi tối)
- `size`: số lượng người tham gia

Bạn hãy thực hiện thống kê mô tả để tìm hiểu về hành vi tip của khách hàng tại nhà hàng

In [4]:
import pandas as pa


In [12]:
tips = pa.read_csv('.\mips.csv')
tips 

Unnamed: 0  total_bill   tip     sex smoker   day    time  size
0             0       16.99  1.01  Female     No   Sun  Dinner     2
1             1       10.34  1.66    Male     No   Sun  Dinner     3
2             2       21.01  3.50    Male     No   Sun  Dinner     3
3             3       23.68  3.31    Male     No   Sun  Dinner     2
4             4       24.59  3.61  Female     No   Sun  Dinner     4
..          ...         ...   ...     ...    ...   ...     ...   ...
239         239       29.03  5.92    Male     No   Sat  Dinner     3
240         240       27.18  2.00  Female    Yes   Sat  Dinner     2
241         241       22.67  2.00    Male    Yes   Sat  Dinner     2
242         242       17.82  1.75    Male     No   Sat  Dinner     2
243         243       18.78  3.00  Female     No  Thur  Dinner     2

[244 rows x 8 columns]

1. Đọc dữ liệu


In [13]:
# Your code here
tips

Unnamed: 0  total_bill   tip     sex smoker   day    time  size
0             0       16.99  1.01  Female     No   Sun  Dinner     2
1             1       10.34  1.66    Male     No   Sun  Dinner     3
2             2       21.01  3.50    Male     No   Sun  Dinner     3
3             3       23.68  3.31    Male     No   Sun  Dinner     2
4             4       24.59  3.61  Female     No   Sun  Dinner     4
..          ...         ...   ...     ...    ...   ...     ...   ...
239         239       29.03  5.92    Male     No   Sat  Dinner     3
240         240       27.18  2.00  Female    Yes   Sat  Dinner     2
241         241       22.67  2.00    Male    Yes   Sat  Dinner     2
242         242       17.82  1.75    Male     No   Sat  Dinner     2
243         243       18.78  3.00  Female     No  Thur  Dinner     2

[244 rows x 8 columns]

2. Xoá cột `Unnamed: 0`


In [14]:
# Your code here
tips.drop('Unnamed: 0',axis=1)

total_bill   tip     sex smoker   day    time  size
0         16.99  1.01  Female     No   Sun  Dinner     2
1         10.34  1.66    Male     No   Sun  Dinner     3
2         21.01  3.50    Male     No   Sun  Dinner     3
3         23.68  3.31    Male     No   Sun  Dinner     2
4         24.59  3.61  Female     No   Sun  Dinner     4
..          ...   ...     ...    ...   ...     ...   ...
239       29.03  5.92    Male     No   Sat  Dinner     3
240       27.18  2.00  Female    Yes   Sat  Dinner     2
241       22.67  2.00    Male    Yes   Sat  Dinner     2
242       17.82  1.75    Male     No   Sat  Dinner     2
243       18.78  3.00  Female     No  Thur  Dinner     2

[244 rows x 7 columns]

3. Cho biết dữ liệu có bao nhiêu dòng, bao nhiêu thuộc tính.


In [21]:
# Your code here
print("So dong- so cot data", tips.shape) #  244 dong, 8 cot             
tips.info() # 8 thuoc tinh,

So dong- so cot data (244, 8)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 244 entries, 0 to 243
Data columns (total 8 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Unnamed: 0  244 non-null    int64  
 1   total_bill  244 non-null    float64
 2   tip         244 non-null    float64
 3   sex         244 non-null    object 
 4   smoker      244 non-null    object 
 5   day         244 non-null    object 
 6   time        244 non-null    object 
 7   size        244 non-null    int64  
dtypes: float64(2), int64(2), object(4)
memory usage: 15.4+ KB


4. Trong các thuộc tính, thuộc tính nào là thuộc tính phân loại, thuộc tính nào là thuộc tính định lượng


In [22]:
# Your code here
tips.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 244 entries, 0 to 243
Data columns (total 8 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Unnamed: 0  244 non-null    int64  
 1   total_bill  244 non-null    float64
 2   tip         244 non-null    float64
 3   sex         244 non-null    object 
 4   smoker      244 non-null    object 
 5   day         244 non-null    object 
 6   time        244 non-null    object 
 7   size        244 non-null    int64  
dtypes: float64(2), int64(2), object(4)
memory usage: 15.4+ KB


Thuộc tính phân loại (categorical data): sex; smoker; day; time;  size: có thể xem là thuộc tính phân loại vì số lượng giá trị ít               ..............Thuộc tính định lượng (numerical data/ quantitative data): total_bill; tip; size

5. Có bao nhiêu nam, bao nhiêu nữ. Tỷ lệ bao nhiêu?


In [23]:
# Your code here
tips['sex'].value_counts()

sex
Male      157
Female     87
Name: count, dtype: int64

In [26]:
a=tips['sex'].value_counts()
b=tips['sex'].count()
print("Tỉ lệ nam nữ", a/b * 100)

Tỉ lệ nam nữ sex
Male      64.344262
Female    35.655738
Name: count, dtype: float64


6. Có bao nhiêu người hút thuốc. Tỷ lệ bao nhiêu?


In [35]:
# Your code here
a = tips[tips['smoker']=='Yes'].count()
b = tips[tips['smoker']=='No'].count()
c = a/b * 100
print("Tỉ lệ khách hàng không hút thuốc", c)
print("Tỉ lệ khách hàng hút thuốc", 100 -c)


Tỉ lệ khách hàng không hút thuốc Unnamed: 0    61.589404
total_bill    61.589404
tip           61.589404
sex           61.589404
smoker        61.589404
day           61.589404
time          61.589404
size          61.589404
dtype: float64
Tỉ lệ khách hàng hút thuốc Unnamed: 0    38.410596
total_bill    38.410596
tip           38.410596
sex           38.410596
smoker        38.410596
day           38.410596
time          38.410596
size          38.410596
dtype: float64


Cách 2:

In [37]:
# Định nghĩa hàm
def thong_ke_thuoc_tinh(data, att):
    '''
        data: a DataFrame
        att: a string of an attribute of data
    '''
    print('Thống kê số lượng: ')
    print(data[att].value_counts())
    
    print('\n\nTỷ lệ: ')
    tyle = round((data[att].value_counts() / data[att].count()) * 100, 2)
    print(tyle)

In [39]:
thong_ke_thuoc_tinh(tips, 'smoker')


Thống kê số lượng: 
smoker
No     151
Yes     93
Name: count, dtype: int64


Tỷ lệ: 
smoker
No     61.89
Yes    38.11
Name: count, dtype: float64


7. Ngày nào trong tuần khách hàng hay đến dùng bữa nhất. Tỷ lệ bao nhiêu?


In [40]:
# Your code here
thong_ke_thuoc_tinh(tips, 'day')


Thống kê số lượng: 
day
Sat     87
Sun     76
Thur    62
Fri     19
Name: count, dtype: int64


Tỷ lệ: 
day
Sat     35.66
Sun     31.15
Thur    25.41
Fri      7.79
Name: count, dtype: float64


8. Nhà hàng phục vụ bữa ăn nào nhiều nhất (Breakfast, Lunch, Dinner)?


In [41]:
# Your code here
thong_ke_thuoc_tinh(tips, 'time')


Thống kê số lượng: 
time
Dinner    176
Lunch      68
Name: count, dtype: int64


Tỷ lệ: 
time
Dinner    72.13
Lunch     27.87
Name: count, dtype: float64


9. Nhóm khách hàng dùng bữa tại nhà hàng nhiều nhất là bao nhiêu người?


In [42]:
# Your code here
thong_ke_thuoc_tinh(tips, 'size')


Thống kê số lượng: 
size
2    156
3     38
4     37
5      5
1      4
6      4
Name: count, dtype: int64


Tỷ lệ: 
size
2    63.93
3    15.57
4    15.16
5     2.05
1     1.64
6     1.64
Name: count, dtype: float64


10. Trung bình trong một bữa ăn nhà hàng phục vụ mấy khách?

In [45]:
# Your code here
print("Trung bình trong một bữa ăn nhà hàng phục vụ mấy khách?", tips['size'].mean())

Trung bình trong một bữa ăn nhà hàng phục vụ mấy khách? 2.569672131147541


---